File: sources/leetcode/26-remove-duplicates-from-sorted-array/main.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums)
    {
        size_t size = nums.size();
        if(size == 0) return 0;
        
        size_t copyTo = 1;
        for(size_t i = 1;i < size;++i)
        {
            if(nums[i-1] != nums[i])
            {
                nums[copyTo] = nums[i];    
                ++copyTo;
            }
        }
        
        return copyTo;
    }
};

File: sources/leetcode/26-remove-duplicates-from-sorted-array/main.rs
impl Solution
{
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32
    {
        if(nums.len() == 0) { return 0; }
        
        let mut copyto: usize = 1;
        let mut i: usize = 1;
        
        while(i < nums.len())
        {
            if(nums[i] != nums[i-1])
            {
                nums[copyto] = nums[i];    
                copyto+=1;
            }
            i+=1;
        }
        
        return copyto as i32;
    }
}